lib/terminal/terminal_interpreter.dart

```diff
@@ -0,0 +1,457 @@
+import 'dart:math';
+
+import 'package:pixel_prompt/pixel_prompt.dart';
+
+enum TerminalState { csi, normal }
+
+class TerminalInterpreter {
+  int currentLine = 0;
+  int currentColumn = 0;
+
+  AnsiColorType? _foregroundColor;
+  AnsiColorType? _backgroundColor;
+
+  final Set<FontStyle> _fontStyles = {};
+
+  int lines;
+  int columns;
+
+  List<List<CellState>> bufferState;
```

Comment:
This could be a class that has a 

```dart
final class BufferState {
  CellState operator [](Position position) => ...
}
```
 
So that all the `bufferState[foo][bar].` becomes `bufferState[position]`. This prevents anyone contributing to this package messing up column with line and prevents anyone messing up whether the indexing is zero-based or one-based.

(Also you can initialize it with a width/height in the `BufferState` constructor instead of in the interpreter constructor.)